### ISSUE TITLE
PR title is not updated to indicate newer versions

### ISSUE THREAD

**Issue by froque** [renovatebot/renovate#13882]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.56.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

13.11.1

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Renovate is not updating the title and description for a existing pull request when it updates to a newer version. 

Pull request title and description:
![Renovate PR title and description](https://user-images.githubusercontent.com/8249379/151526409-1d08e187-a539-41d9-b76a-f5e863ecbf56.png)

Actual change merged:
![Renovate PR diff](https://user-images.githubusercontent.com/8249379/151526557-3c1f3354-8463-43ac-982d-60e2956bcf63.png)

Renovate should be able to update the pull request title and description. 

Note: Renovate is running self-hosted on gitlab (also self-hosted), but these screenshots are from bitbucket.org where the git repository is located. 

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by froque** [comment]
logs and minimal reproduction repository added.

**Comment by rarkins** [comment]
Thank you. Here's what seems to be the important part of the logs:

```
INFO: Branch updated (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
       "commitSha": "0679498"
DEBUG: Updating renovate/stability-days status check state to yellow (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
DEBUG: Setting branch status (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
       "context": "renovate/stability-days",
       "state": "yellow"
DEBUG: Branch status pending (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
       "commitSha": "0679498"
```

We see the branch is updated, and the status check set to pending, but then it seems Renovate discontinues working on the branch, such as ensuring PR.

**Comment by StinkyLord** [comment]
assign this to me please

**Comment by khitrenovich** [comment]
I spotted the same issue in Renovate app on GitHub - PR title and description are not updated to reflect the fact that the PR was bumped to the next version. Can it be related?

In the screenshot below, `vm2` was bumped to v3.9.8, but the PR title and description still mention v3.9.7. We spotted that only because we have [linting on PR titles](https://github.com/amannn/action-semantic-pull-request) and the title did not match the only commit in the branch.

<img width="638" alt="image" src="https://user-images.githubusercontent.com/3424762/154302720-1e3daf4b-2c5e-4924-919a-3983e415f9b4.png">

**Comment by rarkins** [comment]
Is it only temporary and gets updated on the next run?

**Comment by khitrenovich** [comment]
I see it was indeed updated 2 hours later (actually, less than half an hour ago). It might be a red herring then. Sorry about that!

**Comment by PhilipAbed** [comment]
@rarkins please assign to this account instead thanx!

**Comment by PhilipAbed** [comment]
@rarkins if thats temporary and after 2 hours it gets updated then is this issue as good as closed? or u still see a bug?

**Comment by rarkins** [comment]
It's still a bug

**Comment by PhilipAbed** [comment]
Hi @froque 
The reproduce is too complicated for me, i'm not sure how to debug this with all the docker runs and machine,  
is it possible to reproduce using a normal repository and normal runs? 
or maybe a way to reproduce using the .devContainer?

Hi @rarkins 
can you please take a look at that reproduce file and tell me if its possible to debug that and how?

I'm currently trying to find the logic behind the problem from code, but its taking much time without debugging, and i might not even be able to find the problem, i need help here.

**Comment by froque** [comment]
The gist is:
 - renovate should create a PR to upgrade a dependency
 - this PR should be kept open, without closing it or merging it
 - a new version of the dependency should be released
 - renovate detects this new version and updates the branch, but fails to update the PR title and description

the flags `stabilityDays` and `prCreation` seem important for this issue

**Comment by rarkins** [comment]
Unfortunately we're doing to need a more specific description of what is being reproduced so that we can determine what/where to debug. Ideally this would be as simple as a single dependency and single branch, then we don't need to selectively debug.

**Comment by PhilipAbed** [comment]
@rarkins  i'm still trying other ways to debug from ide, when i fail ill let u know

**Comment by froque** [comment]
@rarkins I don't understand the reproduction needed change.  The provided repository only contains a single dependency and a single branch. 
What else should I provide ?

**Comment by PhilipAbed** [comment]
@froque hi, the reproduction contains lots of dockers and connection between docker installations, 
it's a bit complicated, so what im trying to do is split it into multiple small steps on my machine, where i can run the docker nexus, and run everything else manually, because in order to locate the problem i need to run in the IDE in debug mode, and go over the lines and see what's up.
just reproducing with running commands is a reproduction, but i cant debug it using these docker commands, the IDE doesnt run docker commands.

so i'm currently trying to understand what you are doing in the docker and trying to imitate it manually on my machine while trying to debug the problem from the IDE.

**Comment by PhilipAbed** [comment]
@rarkins I successfully started debugging!, but i'm facing another issue, the code is not Uploading any pull request and returns with Failed to update warning... so i debugged and i noticed something weird take a look at this, maybe its another issue or a new issue i'm not sure, check this out:
![image](https://user-images.githubusercontent.com/42116482/155511669-cb9a50f0-c0bb-450c-a8ef-3549e53e35d7.png)

the code gave a position "fileReplacePosition" here:
![image](https://user-images.githubusercontent.com/42116482/155511762-bbc25e42-e6e1-46cb-a54a-315190da22e6.png)

what's amazing is that this screen shot is from my OWN project, i wasn't sure if its @froque 's project problem, so i tried on my own discord `StinkyLord/StinkyBot` which also have maven, i think for some reason we are extracting `fileReplacePosition` wrongly? or is it something i'm missing? can you please advise?

**Comment by PhilipAbed** [comment]
after deeper investigation, if i copy the file content to Notepad++ and GOTO offset `position` , it will point directly to the right location!!
but when i do `fileContent.slice(0,position)` it splits into wrong location

**Comment by PhilipAbed** [comment]
after investigating with nabeel, it appears the code is Parsing the file with "\r\n" on windows, but when we do git.show(origin/{branch}/{pathToFile}), the file comes with "\n" only, which causes the problem. i will bypass this then hopefully i will reproduce the issue 100%.

maybe we should open an issue for this case? or is it only because i took a linux project and used docker while running from windows cli/ide?

**Comment by viceice** [comment]
I think we should open an Uber issue for windows. We have some more line ending issues on windows.

**Comment by PhilipAbed** [comment]
yeah i found that when we get the file we read it from the temporary folder, 
but when we are inside update we do git.show(origin/branch/filePath)
![image](https://user-images.githubusercontent.com/42116482/155538081-d32e6a9d-2c78-42d2-bad5-2972b5cc88bc.png)

vs 

![image](https://user-images.githubusercontent.com/42116482/155538157-fc7ed5a1-3e94-4396-8aea-afbc88f208dc.png)

**Comment by PhilipAbed** [comment]
![image](https://user-images.githubusercontent.com/42116482/155877042-3a8cba00-ecf7-426f-bee2-3223213d4c37.png)
what do you mean by Status Checks? what if i dont have status checks or what status checks??

Basically i might have a clue about what's happening, what i'm, thinking right now is we are opening a branch but not a PR, because:  `"prCreation": "not-pending"`

then when we are upgrading again from version 1.1 to 1.2 on the same branch, 
 and trying to create a new PR with the new version, but since it's using the old Branch to create the new change. code gets confused between versions.
 
 Thing is, **im still not able to reproduce** because the code sees this config: `"prCreation": "not-pending"` and doesn't create any PR for me, so anybody can explain to me about this "status check" and if it might cause a problem?

**Comment by PhilipAbed** [comment]
> Thank you. Here's what seems to be the important part of the logs:
> 
> ```
> INFO: Branch updated (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
>        "commitSha": "0679498"
> DEBUG: Updating renovate/stability-days status check state to yellow (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
> DEBUG: Setting branch status (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
>        "context": "renovate/stability-days",
>        "state": "yellow"
> DEBUG: Branch status pending (repository=froque/renovate-pr-update-issue-13882, branch=renovate/com.premiumminds.test-renovate-pr-update-issue-13882-lib-1.x)
>        "commitSha": "0679498"
> ```
> 
> We see the branch is updated, and the status check set to pending, but then it seems Renovate discontinues working on the branch, such as ensuring PR.

the only reason you see this is because of prCreation = "not-pending" and stabilityDays = 3
The self hosted renovate is supposed to pick up the changes after 3 days i guess ( status checks are done for stability days) 
so the actual logs after 3 days aren't showing where we do "Ensuring PR",

I'm not able to reproduce at all using the given information.

what i think happened is this: we send 2 upgrade requests, one for version upgrade to 1.1 and a second for version upgrade to 1.2, but the 1.2 upgrade has already modified the same branch that 1.1 created ( branch that ends with "1.x" and is shared between the two.), 
so basically the upgrade to 1.1 is showing in the PR but the branch update is showing for 1.2.

i tried many ways to reproduce this but since there is `stabilityDays` and `prCreation :"not-pending",` its a bit hard to reproduce
or do i have to wait 3 days and see what happens? because no PRs are created for me before that and i haven't waited that long..
also i tried playing with the stability days and removing these config params completely after a while, still didn't reproduce what i wanted..

help would be appreciated, im stuck here and not sure how to debug this or reach a point where i can know what's up..

**Comment by froque** [comment]
There is a reproduction
> Reproduction repository: https://github.com/froque/renovate-pr-update-issue-13882, check file reproduce-github.sh

**Comment by rarkins** [comment]
Please add a readme to the repo which describes expected vs observed behavior, plus relevant logs
